From 5d284eaf0f6398f5fc6827a0a205f2a5d0469adc Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Thu, 1 Sep 2022 10:53:37 +0300
Subject: [PATCH 090/102] fdt: pcie: Do not skip fdt fixups for Linux

In case of using the 'skip' option for pcie nodes in 'hwconfig',
the fdt fixup should be skipped only in case of U-Boot fdt and
not Linux fdt.

'skip' option only bypasses the U-Boot configuration of SerDes
and PCIe and not Linux. Otherwise, if the Linux fdt is not
configured correctly, including the necessary fixups, the PCIe
controller configuration will fail during Linux boot, when using
the 'skip' option in 'hwconfig'.

Issue: ALB-9200
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hwconfig_fixups.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
index 7276a42102..a2b82480cc 100644
--- a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
+++ b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
@@ -524,18 +524,22 @@ static int skip_dts_nodes_config(struct dts_node *root, int id, bool *skip)
 	enum serdes_dev_type mode;
 	struct dts_node serdes;
 
+	*skip = false;
+
 	ret = node_by_alias(root, &serdes, SERDES_ALIAS_FMT, id);
 	if (ret) {
 		pr_err("Failed to get 'serdes%u' alias\n", id);
 		return ret;
 	}
 
+	/* Do not skip SerDes & PCIe fixup for Linux device tree */
+	if (root->fdt)
+		return 0;
+
 	mode = s32_serdes_get_mode_from_hwconfig(id);
 
-	if (!(mode & SERDES_SKIP)) {
-		*skip = false;
+	if (!(mode & SERDES_SKIP))
 		return 0;
-	}
 
 	printf("Skipping configuration for SerDes%d.\n", id);
 	*skip = true;
@@ -774,7 +778,7 @@ static void disable_serdes_pcie_nodes(struct dts_node *root, u32 id)
 
 static int apply_hwconfig_fixups(bool fdt, void *blob)
 {
-	bool skip;
+	bool skip = false;
 	int ret;
 	unsigned int id;
 	struct dts_node root = {
-- 
2.17.1

